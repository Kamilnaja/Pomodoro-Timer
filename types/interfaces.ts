export interface Registration extends Login {
  email: string;
}

export interface Login {
  login: string;
  password: string;
}

export interface LoginResponse {
  token: string;
}

export interface AuthError {
  code: ErrorCodes;
  message: string;
}

export enum ErrorCodes {
  USER_NOT_FOUND,
  USER_CURRENTLY_EXISTS,
  PASSWORD_INCORRECT,
  OTHER_ERROR,
}
