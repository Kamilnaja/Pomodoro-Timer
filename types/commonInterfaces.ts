export interface SearchResult<T> {
  result: T;
}

export interface ResultInPeriod {
  hasPreviousPeriod: boolean;
  hasNextPeriod: boolean;
}
