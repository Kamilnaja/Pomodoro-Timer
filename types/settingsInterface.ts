import { DisplayDirection } from '../web/src/settings/store/interfaces/settingsInterfaces';

export interface Settings {
  isCookieConsentAccepted: boolean;
  isSoundEnabled: boolean;
  displayDirection: DisplayDirection;
}
