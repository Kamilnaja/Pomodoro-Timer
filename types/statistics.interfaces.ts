interface Result {
  date: string;
  count: number;
}

export default interface StatsSearchResult {
  result: Result[];
}
