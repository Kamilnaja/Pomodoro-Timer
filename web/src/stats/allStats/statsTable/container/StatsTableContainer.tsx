import React from 'react';
import { connect } from 'react-redux';
import { SettingsState, SortDirection } from '../../../../settings/store/interfaces/settingsInterfaces';
import { StatsTableComponent } from '../component/StatsTableComponent';
import { StatsTableContainerProps } from './StatsTableContainerProps';
import { handleGetSettings, handleSaveSettings } from '../../../../settings/store/actions/settingsActions';

class StatsTableContainer extends React.Component<StatsTableContainerProps> {
  componentDidMount() {
    this.props.handleGetSettings();
  }

  toggleSortDirection = (): void => {
    const { settings } = this.props.settingsState;
    this.props.handleSaveSettings({
      ...settings,
      sortDirection: settings.sortDirection === SortDirection.ASC ? SortDirection.DESC : SortDirection.ASC,
    });
  };

  render() {
    return (
      <StatsTableComponent
        pageMonth={this.props.pageMonth}
        pageYear={this.props.pageYear}
        pomodoros={this.props.pomodoros}
        sortDirection={this.props.settingsState.settings.sortDirection}
        toggleSortDirection={this.toggleSortDirection}
      ></StatsTableComponent>
    );
  }
}

const mapStateToProps = (state: { settings: SettingsState }) => {
  const settingsState = state.settings;
  return { settingsState };
};

const mapDispatchToProps = {
  handleSaveSettings,
  handleGetSettings,
};

export default connect(mapStateToProps, mapDispatchToProps)(StatsTableContainer);
