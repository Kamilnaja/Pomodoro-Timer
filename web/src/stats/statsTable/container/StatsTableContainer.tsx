import React from 'react';
import { ErrorComponent } from '../../../shared/error/errorComponent/ErrorComponent';
import { Loader } from '../../../shared/loader/Loader';
import { StatsTableComponent } from '../component/StatsTableComponent';
import { StatsTableContainerProps } from './StatsTableContainerProps';

class StatsTableContainer extends React.Component<StatsTableContainerProps> {
  render() {
    if (this.props.settingsState.isLoading) {
      return <Loader />;
    } else if (this.props.settingsState.error) {
      return <ErrorComponent />;
    } else {
      return (
        <StatsTableComponent
          pageMonth={this.props.pageMonth}
          pageYear={this.props.pageYear}
          pomodoros={this.props.pomodoros}
          settings={this.props.settingsState.settings}
          toggleDisplayDirection={this.props.toggleDisplayDirection}
          toggleDisplayEmptyDays={this.props.toggleDisplayEmptyDays}
        ></StatsTableComponent>
      );
    }
  }
}

export default StatsTableContainer;
