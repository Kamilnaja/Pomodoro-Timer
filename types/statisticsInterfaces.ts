import { SearchResult } from './commonInterfaces';

/*
 * Allows for searching in current year, month or day
 *
 */
export interface SearchDate {
  year: number;
  month: number;
  day?: number;
}

export interface PomodorosDoneInDay {
  date: string;
  count: number;
}

export interface TodayStats {
  result: number;
}

export interface StatsSearchResult {
  pomodoros: PomodorosDoneInDay[];
  accountActiveFrom: Date;
}

export interface TodayStatsResult extends SearchResult<number> {}
