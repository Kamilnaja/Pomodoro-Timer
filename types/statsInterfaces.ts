import { SearchResult } from './commonInterfaces';

/*
 * Allows for searching in current year, month or day
 *
 */
export interface SearchDate {
  year: number;
  month: number;
  day?: number;
}

export interface PomodorosDoneInDay {
  date: string;
  count: number;
}

export interface TodayStats {
  result: number;
}

export interface StatsSearchResult {
  pomodoros: PomodorosDoneInDay[];
  hasPreviousPeriod: boolean;
  hasNextPeriod: boolean;
}

export interface PomodoroSearchResult extends SearchResult<Pomodoro> {}

export interface TodayStatsResult extends SearchResult<number> {}

export interface Pomodoro {
  tagText: string;
  createdAt: string;
}
