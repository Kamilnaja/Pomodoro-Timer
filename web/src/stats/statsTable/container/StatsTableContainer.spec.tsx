import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Settings } from '../../../../../types/settingsInterface';
import { createSettingsState } from '../../../settings/testing/settings.test.data';
import { createPomodorosDoneInDay } from '../../testing/stats.test.data';
import StatsTableContainer from './StatsTableContainer';

describe('StatsTableContainer', () => {
  let wrapper = shallow(
    <StatsTableContainer
      pageMonth={11}
      pageYear={2020}
      pomodoros={createPomodorosDoneInDay()}
      settingsState={createSettingsState()}
      toggleDisplayDirection={() => {}}
      handleGetSettings={() => {}}
      handleSaveSettings={() => {}}
    />,
  );

  it('Should match snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
