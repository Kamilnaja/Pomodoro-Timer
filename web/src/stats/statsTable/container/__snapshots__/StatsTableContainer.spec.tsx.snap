// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsTableContainer Should match snapshot 1`] = `
<StatsTableComponent
  displayDirection="ASC"
  pageMonth={11}
  pageYear={2020}
  pomodoros={
    Array [
      Object {
        "count": 8,
        "date": "21-02-2021",
      },
      Object {
        "count": 15,
        "date": "20-02-2021",
      },
      Object {
        "count": 16,
        "date": "19-02-2021",
      },
      Object {
        "count": 19,
        "date": "18-02-2021",
      },
      Object {
        "count": 18,
        "date": "17-02-2021",
      },
      Object {
        "count": 11,
        "date": "16-02-2021",
      },
      Object {
        "count": 16,
        "date": "15-02-2021",
      },
      Object {
        "count": 13,
        "date": "14-02-2021",
      },
      Object {
        "count": 14,
        "date": "13-02-2021",
      },
      Object {
        "count": 9,
        "date": "12-02-2021",
      },
      Object {
        "count": 11,
        "date": "11-02-2021",
      },
      Object {
        "count": 13,
        "date": "10-02-2021",
      },
      Object {
        "count": 13,
        "date": "09-02-2021",
      },
      Object {
        "count": 18,
        "date": "08-02-2021",
      },
      Object {
        "count": 14,
        "date": "07-02-2021",
      },
      Object {
        "count": 16,
        "date": "06-02-2021",
      },
      Object {
        "count": 13,
        "date": "05-02-2021",
      },
      Object {
        "count": 12,
        "date": "04-02-2021",
      },
      Object {
        "count": 16,
        "date": "03-02-2021",
      },
      Object {
        "count": 15,
        "date": "02-02-2021",
      },
      Object {
        "count": 17,
        "date": "01-02-2021",
      },
    ]
  }
  toggleDisplayDirection={[Function]}
/>
`;
